src/apps/frontend/pages/tasks/task-header.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import toast from 'react-hot-toast';
+
+import {
+  Button, Flex, FlexItem, HeadingLarge,
+} from '../../components';
+import { AsyncError } from '../../types';
+import { TaskOperationType } from '../../types/task';
+
+import TaskItemCard from './task-item-card';
+import TaskModal from './task-modal';
+import useTaskForm from './tasks-form.hook';
+
+interface TaskHeaderProps {
+  onError?: (error: AsyncError) => void;
+}
+
+const TaskHeader: React.FC<TaskHeaderProps> = ({ onError }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const onSuccess = () => {
+    toast.success('Task has been added successfully');
+    setIsModalOpen(false);
+  };
+
+  const { addTaskFormik, setFormikFieldValue } = useTaskForm({
+    onError,
+    onSuccess,
+  });
+
+  return (
+    <TaskItemCard>
```

Comment:
It is not clear why `TaskHeader` which has no task item uses `TaskItemCard`. This is odd

Reply:
TaskItemcard is essentially a background card for both the header and the task section.

Reply:
Fixed by renaming the file to TaskCard.